src/index.ts

```diff
@@ -151,5 +135,34 @@ function handleStaticGeneration (nuxt: any) {
   })
 }
 
+function prepareLocalProvider ({ nuxt, options }, providerOptions) {
+  // Default port
+  const defaultPort =
+   process.env.PORT ||
+   process.env.npm_package_config_nuxt_port ||
+   (options.server && options.server.port) ||
+   3000
+
+  // Default host
+  let defaultHost =
+   process.env.HOST ||
+   process.env.npm_package_config_nuxt_host ||
+   (options.server && options.server.host) ||
+   'localhost'
+
+  /* istanbul ignore if */
+  if (defaultHost === '0.0.0.0') {
+    defaultHost = 'localhost'
+  }
+
+  // Default prefix
+  const prefix = '/'
+
+  return defu(providerOptions, {
+    baseURL: `http://${defaultHost}:${defaultPort}${prefix}`,
```

Comment:
- baseURL in browser may differ from `HOST` when deployed to production
- For server-side, we may directly listen to a random port 

In conclusion, it would be best if at least support two baseURL options and improve later

Reply:
I have created `internalBaseURL` option for server-side usage. [`4bd3543` (#37)](https://github.com/nuxt/image/pull/37/commits/4bd3543aa2bb5d9bc0addaf8e73be39c2d8cb181)